examples/rig-kafka-pub-sub-example/example/README.md

```diff
@@ -0,0 +1,309 @@
+### Overview
+
+We send a POST request to the Reactive Gateway (RIG), which publishes to Kafka, then that Kafka topic is consumed by a Springboot application, which does a simple data transformation and publishes back onto a topic, which is consumed by RIG and published to the front end.
+
+This demo is currently showing this flow from the commandline.
+
+### Setting Up The Environment 
+Clone and run the docker-starter repo. This creates an instance of Kafka, Zookeeper, and RIG.
+
+```bash
+cd docker-starter
+./start.sh
+```
+
+Notes:
+- Topics for kafka are set at the `KAFKA_SOURCE_TOPICS` flag:
+
+```yml
+  reactive-interaction-gateway:
+    container_name: reactive-interaction-gateway
+    image: accenture/reactive-interaction-gateway:2.1.1
+    environment:
+      - LOG_LEVEL=debug
+      - KAFKA_SOURCE_TOPICS=rig,rig-consumed
+      - PROXY_KAFKA_REQUEST_TOPIC=rig
+      - KAFKA_BROKERS=kafka:9092
+      - API_HTTP_PORT=7010
+      - INBOUND_PORT=7000
+      - PROXY_CONFIG_FILE=proxy/rig-proxy.json
+```
+
+- The internal and the external ports for Kafka are different. 
+
+In the dockerfile
+```yml
+ kafka:
+    image: confluentinc/cp-kafka:5.0.0
+    container_name: kafka
+    ports:
+      - 9092:9092 # container communication
+      - 9094:9094 # localhost communication
+    environment:
+      KAFKA_ZOOKEEPER_CONNECT: zk:2181
+      KAFKA_ADVERTISED_LISTENERS: INSIDE://kafka:9092,OUTSIDE://localhost:9094
+      KAFKA_LISTENERS: INSIDE://0.0.0.0:9092,OUTSIDE://0.0.0.0:9094
+```
+
+You'll see below when we ssh into the container, from the CLI there we use 9092.
+
+However our Springboot application yaml is the public facing 9094
+
+```yml
+spring:
+  profiles: local
+  kafka:
+    properties:
+      security.protocol: PLAINTEXT
+    consumer:
+      bootstrap-servers: 127.0.0.1:9094
+    producer:
+      bootstrap-servers: 127.0.0.1:9094
+```
+
+### Starting the Java App
+
+Build the app
+```bash
+mvn clean package
+```
+
+Run the app 
+```bash
+java -jar target/example-0.0.1-SNAPSHOT.jar
```

Comment:
Any reason to not use Docker image?

Reply:
No, other than laziness and lack of fluency in Docker. I am working on this and have added it to the Dockerfile, but I can't get the Springboot java image to interact with Kafka. Any ideas?

Reply:
Docker Compose file

Reply:
update - I got it to work